envbuilder.go

```diff
@@ -1490,9 +1490,15 @@ func initDockerConfigJSON(logf log.Func, magicDir constants.MagicDir, dockerConf
 		return noop, fmt.Errorf("write docker config: %w", err)
 	}
 	logf(log.LevelInfo, "Wrote Docker config JSON to %s", cfgPath)
+	oldDockerConfig := os.Getenv("DOCKER_CONFIG")
+	os.Setenv("DOCKER_CONFIG", magicDir.String())
+	logf(log.LevelInfo, "Set DOCKER_CONFIG to %s", magicDir.String())
```

Comment:
Nit: for the absence of doubt, it might be best to read the newly-set env var instead of what it _should_ (and almost certainly _will_) be.